src/Logger/Logger.cc

```diff
@@ -6,16 +6,14 @@
 
 #include "Logger.h"
 
+#include "Main/ProgramSettings.h"
+
 #include <string>
 
 namespace carta {
 namespace logger {
 
-static bool log_protocol_messages(false);
-
-void InitLogger(bool no_log_file, int verbosity, bool log_performance, bool log_protocol_messages_, fs::path user_directory) {
-    log_protocol_messages = log_protocol_messages_;
-
+void InitLogger(const ::carta::ProgramSettings& settings) {
```

Comment:
My mistake; I didn't realise in my previous review that this was a collection of functions and not a class.

This should be refactored to be closer to the original solution: restore the static `log_protocol_messages` variable and set it in this function using the value from the settings object. But don't pass the settings object in as a parameter; call `GetInstance` once at the start of the function to create a local reference. The remaining functions should be reverted to use the static variable, as before.